Modules/LDAPObject.c

```diff
@@ -1162,7 +1162,7 @@ l_ldap_search_ext( LDAPObject* self, PyObject* args )
     int msgid;
     int ldaperror;
 
-    if (!PyArg_ParseTuple( args, "sis|OiOOdi",
+    if (!PyArg_ParseTuple( args, "zis|OiOOdi",
```

Comment:
Just move the ``|`` to the left and initialize the targets with NULL/0.

Reply:
We can make all arguments optional, too. ``z`` is correct to accept None and map it to NULL string.

Reply:
This is only called from python-ldap Python code and that will already pass the right arguments in (with a default), so the separator can stay where it is?